### DISCUSSION TITLE
alternate format?

### DISCUSSION THREAD

**Discussion by Laertees** [shader-park/shader-park-core#90] [General]
Hi, I am playing around with the shadertoy to touch designer translator and some examples work perfectly, but the 
shaders that looks more like "real" GLSL doesnt translate (such as https://shaderpark.com/sculpture/-Lhla0Vw2uP0RFTlyXn8)

Am I missing something fundamental?

I am very new at this so please excuse me if this is a F.A.Q.

kind regards, 
/Lars

**Comment by torinmb** [comment]
Hi Lars,

At the moment there isn't a way to convert the raw GLSL Shader Park examples into the correct format for TouchDesigner. Shader Park started off as a GLSL helper library, and then we realized how helpful it was to abstract it into JavaScript (which uses our GLSL library under the hood). At the moment the TouchDesigner Plugin only supports the more recent Javascript format.

One thing to note is that we've created the ability to bind GLSL into javascript functions https://docs.shaderpark.com/references-js/glsl/glslFunc.html which is supported in the TD plugin. 

If you're hoping to bring in the font, you'll need to include this texture ![msdf-left-align](https://user-images.githubusercontent.com/6014011/200135429-83295455-8c26-434b-84ef-fa99e0de26c1.png) as a uniform in TouchDesinger. (note getting the font to work in TD would require a fair amount of prior shader knowledge). This is something that I'm interested in incorporating to the js library at some point also.

**Reply by torinmb** [reply]
It actually does look like we included the ability to convert a GLSL example in the core library to TouchDesigner's format. I just haven't incorporated it into the TD component.
https://github.com/shader-park/shader-park-core/blob/6e1fa04eaf7e2bae96a86ebd534ac3d878fabdd6/targets/touchDesigner.js#L177

**Comment by Laertees** [comment]
thanks for the quick reply.
I think getting the font to works is probably above my level of "expertise" at the moment :) 

but this is a supercool tool and the ability to play around here and then output it to TD GLSL is amazing.

**Reply by torinmb** [reply]
Hey! I added the ability to render the GLSL shaders in the TD component. <img width="569" alt="Screen Shot 2022-11-07 at 2 33 24 PM" src="https://user-images.githubusercontent.com/6014011/200398901-69bc30e5-935e-4827-b853-4baa2bddbf33.png">

Here's a working example of loading in the font. https://drive.google.com/file/d/12WXHDTVPCx1jS_5f7Q1K9nAF7u8Vmx_Y/view?usp=share_link

Notice that I'm passing a reference from the font texture into the shader_material.

**Reply by Laertees** [reply]
Amazing! 
Will try as soon as possible!!
